### PR TITLE
Electric Welder & Cell Component

### PR DESCRIPTION
**Pull request by thgvr**
## About The Pull Request
Mostly port of https://github.com/Skyrat-SS13/Skyrat-tg/pull/4900
Electric welder from Nova Sector

Removes the old experimental welder and replaces it with a battery-operated welding tool. 
image
Sprite base by rye and finished by me
Also adds a cell component, allowing an easy way to add a functional cell to any item.

## Changelog

:cl:
add: Added a new electric welder - buy it from the outpost.
del: Removed experimental welder
/:cl:

### CODE REVIEW

**Comment by rye-rice** [comment]
STG remind me tommorw about this

**Comment by rye-rice** [comment]
oh my god remind me on firday

**Comment by Sun-Soaked** [comment]
I think fallcon ported the tg version of this cell component- we have it under components/melee/charged

Check it out and see if you can generalize it for use in tools

**Comment by thgvr** [comment]
Weapons should just use this

File: code/game/objects/items/storage/belt.dm [hunk 1]
```diff
@@ -87,7 +87,6 @@
 /obj/item/storage/belt/utility/chief/full/PopulateContents()
 	new /obj/item/screwdriver/power(src)
 	new /obj/item/crowbar/power(src)
-	new /obj/item/weldingtool/experimental(src)//This can be changed if this is too much
```
Comment (MarkSuckerberg) [inline code/game/objects/items/storage/belt.dm:90]: dunno if this is even spawned anywhere but I feel like this should just be replaced with a normal highcap welder probably

File: code/game/objects/items/tools/electric_weldingtool.dm [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	. = ..()
+	. -= "It contains [get_fuel()] unit\s of fuel out of [max_fuel]."
+
+// This is what uses fuel in the parent. We override it here to not use fuel
+/obj/item/weldingtool/electric/use(used = 0)
+	return isOn()
+
+/obj/item/weldingtool/electric/examine()
+	. = ..()
+	. += "[src] is currently [powered ? "powered" : "unpowered"]."
```
Comment (MarkSuckerberg) [inline code/game/objects/items/tools/electric_weldingtool.dm]: these two examine blocks should be one

File: code/datums/components/cell_component.dm [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+		return
+
+	if(!istype(inserting_item, /obj/item/stock_parts/cell))
+		return
+
+	if(inserted_cell) //No quickswap compatibility
+		to_chat(user, span_danger("There is already a cell in [equipment]!"))
+		return
+
+	to_chat(user, span_notice("You connect [inserting_item] onto [equipment]."))
```
Comment (MarkSuckerberg) [inline code/datums/components/cell_component.dm:177]:
```suggestion
	to_chat(user, span_notice("You add [inserting_item] to [equipment]."))
```

File: code/datums/components/cell_component.dm [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+	playsound(equipment,  'sound/weapons/magin.ogg', 40, TRUE)
```
Comment (MarkSuckerberg) [inline code/datums/components/cell_component.dm]:
```suggestion
	playsound(equipment, 'sound/weapons/magin.ogg', 40, TRUE)
```

File: code/game/objects/items/tools/electric_weldingtool.dm [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+		return
+
+	if(!(item_use_power(power_use_amount) & COMPONENT_POWER_SUCCESS))
+		switched_off()
+		return
+
+// We don't need to know how much fuel it has, because it doesn't use any.
+/obj/item/weldingtool/electric/examine(mob/user)
+	. = ..()
+	. -= "It contains [get_fuel()] unit\s of fuel out of [max_fuel]."
```
Comment (MarkSuckerberg) [inline code/game/objects/items/tools/electric_weldingtool.dm]:
```suggestion
/obj/item/weldingtool/electric/examine(mob/user)
	. = ..()
	//Overwrite the last entry, which normally shows welder fuel usage
	.[length(.)] = "[src] is currently [powered ? "powered" : "unpowered"]."
``` 
I hate every way to do this. inheritance was a mistake

**Review by MarkSuckerberg — DISMISSED** [review]

File: code/game/objects/items/storage/belt.dm (hunk 1 above)
Comment (thgvr) [reply code/game/objects/items/storage/belt.dm:90]: I'm not sure it's used but yeah - must have missed it

File: code/game/objects/items/tools/electric_weldingtool.dm (hunk 2 above)
Comment (thgvr) [reply code/game/objects/items/tools/electric_weldingtool.dm]: Wonder why it was written like this I didn't even notice

File: code/datums/components/cell_component.dm (hunk 3 above)
Comment (thgvr) [reply code/datums/components/cell_component.dm:177]: Maybe not exactly this, right now it does
`you connect the cell onto the electric welder`
maybe `you connect the cell to the electric welder` will be what i'll do

Comment (MarkSuckerberg) [reply code/datums/components/cell_component.dm:177]: I was considering that too yeah. works

**Comment by Burning02** [comment]
Just saying this here, this PR removed the welder CE belts had, and that one jungle zombie ruin with one for some reason

**Comment by thgvr** [comment]
will fix

**Comment by Sun-Soaked** [comment]
> will fix

idle horror sting # 48 (warner bros media sound effect collection)

**Review by MarkSuckerberg — APPROVED** [review]
yeag looks good

**Review by Sun-Soaked — APPROVED** [review]
heart emonji

**Pull request merged**